### PR TITLE
Add support for webhook configuration.

### PR DESCRIPTION
**Pull request by cube2222**

### CODE REVIEW

**Comment by cube2222** [comment]
[Add webhooks support to the Terraform provider](https://trello.com/c/cyHxYzDm/102-add-webhooks-support-to-the-terraform-provider)

File: spacelift/e2e/stack_webhook_test.go [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+`,
+			Check: resource.ComposeTestCheckFunc(
+				// Test resource.
+				resource.TestCheckResourceAttr("spacelift_stack_webhook.webhook", "stack_id", "babys-first-stack"),
+				resource.TestCheckResourceAttr("spacelift_stack_webhook.webhook", "id", "babys-first-webhook"),
+				resource.TestCheckResourceAttr("spacelift_stack_webhook.webhook", "deleted", "false"),
+				resource.TestCheckResourceAttr("spacelift_stack_webhook.webhook", "enabled", "true"),
+				resource.TestCheckResourceAttr("spacelift_stack_webhook.webhook", "endpoint", "localtest.me/amazin"),
+				resource.TestCheckResourceAttr("spacelift_stack_webhook.webhook", "secret", "test_secret2"),
+
```
Comment (patrycjazawadka) [inline spacelift/e2e/stack_webhook_test.go:59]: // Test data

File: spacelift/data_stack_webhook.go [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+	webhookIndex := -1
+	for i, webhook := range stack.Integrations.Webhooks {
+		if webhook.ID == webhookID {
+			webhookIndex = i
+			break
+		}
+	}
+	if webhookIndex == -1 {
+		d.SetId("")
+		return nil
```
Comment (patrycjazawadka) [inline spacelift/data_stack_webhook.go:75]: i think you should throw error here

File: spacelift/resource_stack_webhook.go [hunk 3]
```diff
@@ -0,0 +1,181 @@
...
+	webhookIndex := -1
+	for i, webhook := range stack.Integrations.Webhooks {
+		if webhook.ID == webhookID {
+			webhookIndex = i
+			break
+		}
+	}
+	if webhookIndex == -1 {
+		d.SetId("")
+		return nil
```
Comment (patrycjazawadka) [inline spacelift/resource_stack_webhook.go:120]: here too

File: spacelift/data_stack_webhook.go (hunk 2 above)
Comment (marcinwyszynski) [reply spacelift/data_stack_webhook.go:75]: No, this is just Terraform's way of indicating "resource not found, need to create it".

**Pull request merged**